### PR TITLE
Add device authentication operation

### PR DESCRIPTION
**Pull request by esdrasjnr**
What does this implement/fix? Explain your changes
---------------------------------------------------
This patch implements the device auth command operation in the client side.

Where has this been tested?
---------------------------

**Operating System/Platform:** Ubuntu 18.04.4

**Go Version:** 1.13.6

### CODE REVIEW

**Comment by joaoaneto** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- pkg/thing/entities/err_thing_forbidden.go  1
- pkg/thing/delivery/amqp/client.go  2
- pkg/thing/interactors/auth_thing.go  6
- pkg/thing/handler/amqp/msg_handler.go  3
- pkg/thing/interactors/auth_thing_test.go  5
- pkg/thing/entities/err_thing_not_found.go  1
         

Complexity decreasing per file
==============================
+ pkg/thing/interactors/register_thing.go  -1
         

Clones added
============
- pkg/thing/interactors/auth_thing_test.go  5
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/joaoaneto/knot-babeltower/pullRequest?prid=5017969&bid=16543340)

**Comment by joaoaneto** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- pkg/thing/entities/err_thing_forbidden.go  1
- pkg/thing/interactors/auth_thing.go  6
- pkg/thing/entities/err_thing_not_found.go  1
- pkg/thing/handler/amqp/msg_handler.go  2
- pkg/thing/interactors/auth_thing_test.go  5
         

Clones added
============
- pkg/thing/interactors/auth_thing_test.go  5
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/joaoaneto/knot-babeltower/pullRequest?prid=5017969&bid=16543340)

File: pkg/thing/handler/amqp/msg_handler.go [hunk 1]
```diff
@@ -103,6 +103,21 @@ func (mc *MsgHandler) handleListDevices(authorization string) error {
 	return mc.thingInteractor.List(authorization)
 }
 
+func (mc *MsgHandler) handleAuthThing(body []byte, authorization string) error {
```
Comment (vitbaq) [inline pkg/thing/handler/amqp/msg_handler.go]: Change to handleAuthDevices

**Review by vitbaq — CHANGES_REQUESTED** [review]

**Comment by joaoaneto** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- pkg/thing/handler/amqp/msg_handler.go  2
- pkg/thing/interactors/auth_thing.go  6
- pkg/thing/entities/err_thing_not_found.go  1
- pkg/thing/entities/err_thing_forbidden.go  1
- pkg/thing/interactors/auth_thing_test.go  5
         

Clones added
============
- pkg/thing/interactors/auth_thing_test.go  5
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/joaoaneto/knot-babeltower/pullRequest?prid=5017969&bid=16543340)

File: pkg/thing/handler/amqp/msg_handler.go (hunk 1 above)
Comment (esdrasjnr) [reply pkg/thing/handler/amqp/msg_handler.go]: @vitbaq Done!

**Comment by netoax** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- pkg/thing/entities/err_thing_not_found.go  1
- pkg/thing/interactors/auth_thing.go  6
- pkg/thing/interactors/auth_thing_test.go  5
- pkg/thing/handler/amqp/msg_handler.go  2
- pkg/thing/entities/err_thing_forbidden.go  1
         

Clones added
============
- pkg/thing/interactors/auth_thing_test.go  5
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/joaoaneto/knot-babeltower/pullRequest?prid=5017969&bid=16543340)

**Comment by netoax** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- pkg/thing/interactors/auth_thing.go  6
- pkg/thing/interactors/auth_thing_test.go  5
- pkg/thing/entities/err_thing_forbidden.go  1
- pkg/thing/handler/amqp/msg_handler.go  2
- pkg/thing/entities/err_thing_not_found.go  1
         

Clones added
============
- pkg/thing/interactors/auth_thing_test.go  5
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/joaoaneto/knot-babeltower/pullRequest?prid=5017969&bid=16543340)

**Review by vitbaq — APPROVED** [review]

**Pull request merged**